src/core/src/op/cos.cpp

```diff
@@ -2,81 +2,68 @@
+struct Evaluate : ov::element::NoAction<bool> {
+    using ov::element::NoAction<bool>::visit;
+
+    template <element::Type_t ET>
+    static result_type visit(const Tensor& arg0, Tensor& out, const size_t count) {
+        using T = typename element_type_traits<ET>::value_type;
+        reference::cos(arg0.data<T>(), out.data<T>(), count);
+        return true;
+    }
+};
+}  // namespace cos
+
+namespace v0 {
+Cos::Cos(const Output<Node>& arg) : UnaryElementwiseArithmetic(arg) {
     constructor_validate_and_infer_types();
 }
 
-bool op::Cos::visit_attributes(AttributeVisitor& visitor) {
+bool Cos::visit_attributes(AttributeVisitor& visitor) {
     OV_OP_SCOPE(v0_Cos_visit_attributes);
     return true;
 }
 
-shared_ptr<Node> op::Cos::clone_with_new_inputs(const OutputVector& new_args) const {
+shared_ptr<Node> Cos::clone_with_new_inputs(const OutputVector& new_args) const {
     OV_OP_SCOPE(v0_Cos_clone_with_new_inputs);
     check_new_args_count(this, new_args);
     return make_shared<Cos>(new_args.at(0));
 }
 
-OPENVINO_SUPPRESS_DEPRECATED_START
-namespace cosop {
-namespace {
-template <element::Type_t ET>
-inline bool evaluate(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count) {
-    using T = typename element_type_traits<ET>::value_type;
-    runtime::reference::cos<T>(arg0->get_data_ptr<ET>(), out->get_data_ptr<ET>(), count);
-    return true;
-}
-
-bool evaluate_cos(const HostTensorPtr& arg0, const HostTensorPtr& out, const size_t count) {
-    bool rc = true;
-    out->set_unary(arg0);
-
-    switch (arg0->get_element_type()) {
-        NGRAPH_TYPE_CASE(evaluate_cos, i32, arg0, out, count);
-        NGRAPH_TYPE_CASE(evaluate_cos, i64, arg0, out, count);
-        NGRAPH_TYPE_CASE(evaluate_cos, u32, arg0, out, count);
-        NGRAPH_TYPE_CASE(evaluate_cos, u64, arg0, out, count);
-        NGRAPH_TYPE_CASE(evaluate_cos, f16, arg0, out, count);
-        NGRAPH_TYPE_CASE(evaluate_cos, f32, arg0, out, count);
-    default:
-        rc = false;
-        break;
-    }
-    return rc;
-}
-}  // namespace
-}  // namespace cosop
-
-bool op::Cos::evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const {
+bool Cos::evaluate(TensorVector& outputs, const TensorVector& inputs) const {
     OV_OP_SCOPE(v0_Cos_evaluate);
-    OPENVINO_SUPPRESS_DEPRECATED_START
-    NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1) && validate_host_tensor_vector(inputs, 1));
-    OPENVINO_SUPPRESS_DEPRECATED_END
-    return cosop::evaluate_cos(inputs[0], outputs[0], shape_size(inputs[0]->get_shape()));
+    using namespace ov::element;
+    return IfTypeOf<i32, i64, u32, u64, f16, f32>::apply<cos::Evaluate>(inputs[0].get_element_type(),
+                                                                        inputs[0],
+                                                                        outputs[0],
+                                                                        shape_size(inputs[0].get_shape()));
```

Comment:
The original check:
 `NGRAPH_CHECK(validate_host_tensor_vector(outputs, 1) && validate_host_tensor_vector(inputs, 1));`
 ensured the size of the input/outputs vector to safely get element at the index.
 
Currently `inputs[0]` and `outputs[0]` are reached without the `TensorVector` size check.

Reply:
This check was not there before and not sure if required at all.
When there is a graph the all inputs should be set and tensor should be created for them correctly.
@ilyachur do we really need check number of input/output tensors? or is done some upper layers when tensors are created?

Reply:
> This check was not there before and not sure if required at all.

The comment is directly related to the change, where such check was before:
<img width="600" alt="image" src="https://github.com/openvinotoolkit/openvino/assets/20084299/ca13ee27-d0ad-4580-ac05-9b8fa12c9e7a">


and currently is on the master:
https://github.com/openvinotoolkit/openvino/blob/cf2c7da568934870c29acc961a4498ff9cbd8d9c/src/core/src/op/cos.cpp#L61-L63

The `evaluate` method is a part of public API of the Node class, also with Python API bindings, 
so it can be used directly by a user, with a TensorVector of any size. 
https://github.com/openvinotoolkit/openvino/blob/a45e5e03c5a2e412447d82989c9d69f7f51f9e41/src/bindings/python/tests/test_graph/test_basic.py#L442-L445
Such check can prevent unexpected issues, especially when the operator has several inputs and outputs.

Reply:
But acos, acosh, atan... and there will be more ops which not implement this check.

Reply:
From my point of view it will be better to check, because technically user can put empty vector of output tensors to the operation, it will be better to create required tensors or throw an exception than get a segfault.